File: src/aranceles/DOM_aranceles.py
import marimo

__generated_with = "0.15.5"
app = marimo.App(width="medium")


@app.cell
def _():
    import marimo as mo
    return (mo,)


@app.cell
def _():
    import altair as alt
    import polars as pl
    import pandas as pd 
    return alt, pd, pl


@app.cell
def _(pl):
    ## Cargamos datos HS2
    df_hs_2d = pl.read_csv("output/hs2_asia_shares_extensivo_intensivo.csv").filter(year=2023)
    df_hs_4d = pl.read_csv("output/asia_shares_extensivo_intensivo.csv").filter(year=2023)

    df_hs_2d = df_hs_2d.with_columns(
        pl.col("export_share_to_usa")*100,
        pl.col("share_on_total_imports")*100,
        pl.col("export_value")/1000_000
    )

    df_hs_4d = df_hs_4d.with_columns(
        pl.col("export_share_to_usa")*100,
        pl.col("share_on_total_imports")*100,
        pl.col("export_value")/1000_000
    )

    ## Subset extensivo 2d
    data_intensivo = df_hs_2d.filter(pl.col("export_rca")>=1.0)

    ## Subset extensivo 2d
    data_extensivo = df_hs_2d.filter(pl.col("export_rca")<1.0)

    ## Subset extensivo 4d
    data_intensivo_4d = df_hs_4d.filter(pl.col("export_rca")>=1.0)

    ## Subset extensivo 4d
    data_extensivo_4d = df_hs_4d.filter(pl.col("export_rca")<1.0)
    return data_extensivo, data_extensivo_4d, data_intensivo, data_intensivo_4d


@app.cell(hide_code=True)
def _(mo):
    mo.md(r"""# HS2""")
    return


@app.cell
def _(alt, data_intensivo, pd):
    intensivo_share_export_usa = alt.Chart(data_intensivo, title=alt.Title("Industrias del Sector Intensivo (HS2)",subtitle="Importaciones de China, Vietnam, Camboya, Malasia, Indonesia (2023)")).mark_circle().encode(
        alt.X('pci:Q').scale(zero=False, padding=10).title("PCI"),
        alt.Y('share_on_total_imports:Q').scale(zero=False, padding=10).title("Razón con respecto al Total de Importaciones [%]"),
        color=alt.Color('product_hs92_name_1d').title("Cluster"),
        size=alt.Size('export_share_to_usa', scale=alt.Scale(range=[50, 1000])).title("Export Share USA [%]"),
    )

    intensivo_points_text = alt.Chart(data_intensivo).mark_text(
        align='left',
        baseline='middle',
        dx=7, 
    ).encode(
        alt.X('pci:Q').scale(zero=False, padding=10),
        alt.Y('share_on_total_imports:Q').scale(zero=False, padding=10),
        text='product_hs92_name_2d'
    )

    percent_50 = alt.Chart(pd.DataFrame({'y': [50]})).mark_rule(color="red", size = 2).encode(y='y')
    eci_dom = alt.Chart(pd.DataFrame({'x': [-0.19]})).mark_rule(color="black", size = 3).encode(x='x')


    intensivo_share_export_usa + intensivo_points_text + percent_50 + eci_dom
    return eci_dom, intensivo_points_text, percent_50


@app.cell
def _(alt, data_intensivo, eci_dom, intensivo_points_text, percent_50):
    intensivo_export_value = alt.Chart(data_intensivo,  title=alt.Title("Industrias del Sector Intensivo (HS2)",subtitle="Importaciones de China, Vietnam, Camboya, Malasia, Indonesia (2023)")).mark_circle().encode(
        alt.X('pci:Q').scale(zero=False, padding=10).title("PCI"),
        alt.Y('share_on_total_imports:Q').scale(zero=False, padding=10).title("Razón con respecto al Total de Importaciones [%]"),
        color=alt.Color('product_hs92_name_1d').title("Cluster"),
        size=alt.Size('export_value', scale=alt.Scale(range=[50, 2000])).title("Export Value (Million USD)"),
    )

    intensivo_export_value + intensivo_points_text + percent_50 + eci_dom
    return


@app.cell
def _(alt, data_extensivo, eci_dom, percent_50):
    extensivo_share_export_usa = alt.Chart(data_extensivo,  title=alt.Title("Industrias del Sector Extensivo (HS2)",subtitle="Importaciones de China, Vietnam, Camboya, Malasia, Indonesia (2023)")).mark_circle().encode(
        alt.X('pci:Q').scale(zero=False, padding=10).title("PCI"),
        alt.Y('share_on_total_imports:Q').scale(zero=False, padding=10).title("Razón con respecto al Total de Importaciones [%]"),
        color=alt.Color('product_hs92_name_1d').title("Cluster"),
        size=alt.Size('export_share_to_usa', scale=alt.Scale(range=[50, 1000])).title("Export Share USA [%]"),
    )

    extensivo_points_text = alt.Chart(data_extensivo).mark_text(
        align='left',
        baseline='middle',
        dx=7, 
    ).encode(
        alt.X('pci:Q').scale(zero=False, padding=10),
        alt.Y('share_on_total_imports:Q').scale(zero=False, padding=10),
        text='product_hs92_name_2d'
    )

    extensivo_share_export_usa + extensivo_points_text + percent_50 + eci_dom
    return (extensivo_points_text,)


@app.cell
def _(alt, data_extensivo, eci_dom, extensivo_points_text, percent_50):
    extensivo_export_value = alt.Chart(data_extensivo, title=alt.Title("Industrias del Sector Extensivo (HS2)",subtitle="Importaciones de China, Vietnam, Camboya, Malasia, Indonesia (2023)")).mark_circle().encode(
        alt.X('pci:Q').scale(zero=False, padding=10).title("PCI"),
        alt.Y('share_on_total_imports:Q').scale(zero=False, padding=10).title("Razón con respecto al Total de Importaciones [%]"),
        color=alt.Color('product_hs92_name_1d').title("Cluster"),
        size=alt.Size('export_value', scale=alt.Scale(range=[50, 2000])).title("Export Value (Million USD)"),
    )

    extensivo_export_value + extensivo_points_text + percent_50 + eci_dom
    return


@app.cell(hide_code=True)
def _(mo):
    mo.md(r"""# HS4""")
    return


@app.cell
def _(alt, data_intensivo_4d, eci_dom, pd, pl):
    plot_data_intensivo_4d = alt.Chart(data_intensivo_4d, title=alt.Title("Industrias del Sector Intensivo (HS4)",subtitle="Importaciones de China, Vietnam, Camboya, Malasia, Indonesia (2023)")).mark_circle(size = 400).encode(
        alt.X('pci:Q').scale(zero=False, padding=10).title("PCI"),
        alt.Y('share_on_total_imports:Q').scale(zero=False, padding=10).title("Razón con respecto al Total de Importaciones [%]"),
        color=alt.Color('product_hs92_name_1d',
              scale=alt.Scale(
                  domain = ["Agriculture", "Chemicals", "Machinery", "Metals", "Minerals", "Stone", "Textiles", "Electronics"],
                  range = ["#1f77b4", "#ff7f0e", "#2ca02c", "#d62728", "#9467bd", "#8c564b", "#e377c2", "#7f7f7f"]
              )
                       ).title("Cluster"),
    )

    plot_data_intensivo_4d_points_text = alt.Chart(
        data_intensivo_4d.filter(
            (pl.col("share_on_total_imports")>40) & 
            (pl.col("pci")>-0.19)
        )
    ).mark_text(
        align='left',
        baseline='middle',
        dx=20, 
        size = 8
    ).encode(
        alt.X('pci:Q').scale(zero=False, padding=10),
        alt.Y('share_on_total_imports:Q').scale(zero=False, padding=10),
        text='product_hs92_name_4d'
    )

    percent_40 = alt.Chart(pd.DataFrame({'y': [40]})).mark_rule(color="red", size = 2).encode(y='y')

    plot_data_intensivo_4d  + plot_data_intensivo_4d_points_text + percent_40 + eci_dom
    return


@app.cell
def _(data_intensivo_4d, pl):
    data_intensivo_4d.filter(
            (pl.col("share_on_total_imports")>40) & 
            (pl.col("pci")>-0.19)
        )
    return


@app.cell
def _(data_intensivo):
    data_intensivo
    return


@app.cell
def _(data_extensivo):
    data_extensivo

    return


@app.cell
def _(data_intensivo_4d):
    data_intensivo_4d
    return


@app.cell
def _(data_extensivo_4d):
    data_extensivo_4d
    return


if __name__ == "__main__":
    app.run()

File: src/aranceles/DOM_aranceles.R
# Libraries
library(ggplot2)
library(dplyr)
#library(hrbrthemes)
library(viridis)
library(ggrepel)

setwd("/home/milo/Documents/egtp/iniciativas/InvESt_complexity/src/aranceles/output")

## Cargamos datos HS2
df_hs_2d = read.csv("hs2_asia_shares_extensivo_intensivo.csv")
df_hs_4d = read.csv("asia_shares_extensivo_intensivo.csv")

df_hs_2d$export_share_to_usa <- df_hs_2d$export_share_to_usa*100
df_hs_2d$share_on_total_imports <- df_hs_2d$share_on_total_imports*100
df_hs_2d$export_value <- df_hs_2d$export_value/1000000
df_hs_2d$export_value_to_usa <- df_hs_2d$export_value_to_usa/1000000

df_hs_4d$export_share_to_usa <- df_hs_4d$export_share_to_usa*100
df_hs_4d$share_on_total_imports <- df_hs_4d$share_on_total_imports*100
df_hs_4d$export_value <- df_hs_4d$export_value/1000000
df_hs_4d$export_value_to_usa <- df_hs_4d$export_value_to_usa/1000000

data_intensivo = subset(df_hs_2d, year == 2023 & export_rca >=1.0)
data_extensivo = subset(df_hs_2d, year == 2023 & export_rca <1.0)

data_intensivo_4d = subset(df_hs_4d, year == 2023 & export_rca >=1.0)
data_extensivo_4d = subset(df_hs_4d, year == 2023 & export_rca <1.0)

ECI_pais = -0.292
umbral_importaciones = 50

### INTENSIVO
## Export value to USA
ggplot(data_intensivo, 
       aes(x = pci, y = share_on_total_imports, size = export_value_to_usa, fill = product_hs92_name_1d, label = product_hs92_name_2d)) +
  geom_point(alpha = .5, 
             shape = 21) +
  geom_text(nudge_x = 0.1, nudge_y = 0.1, size = 4, color = "black")  +
  scale_size_continuous(range = c(1, 16)) +
  labs(title = "Industrias del Sector Intensivo (HS2)",
       subtitle = "Importaciones de China, Vietnam, Camboya, Malasia, Indonesia (2023)",
       x = "PCI",
       y = "Razón con respecto al Total de Importaciones [%]",
       size = "Export value to USA (Million USD)",
       fill = "Cluster") +
  theme_minimal()  +
  guides(fill=guide_legend(override.aes=list(size=6))) + 
  geom_hline(yintercept=50,  color = "red", size = 1.5) + 
  geom_vline(xintercept = ECI_pais, color = "black", size=1.5)

ggsave("../GTM/hs2_export_value_to_usa_intensivo.png",dpi = 300,  width = 1167/90, height = 575/90,  bg = 'white')

## Total export value
ggplot(data_intensivo, 
       aes(x = pci, y = share_on_total_imports, size = export_value, fill = product_hs92_name_1d, label = product_hs92_name_2d)) +
  geom_point(alpha = .5, 
             shape = 21) +
  geom_text(nudge_x = 0.1, nudge_y = 0.1, size = 4, color = "black")  +
  scale_size_continuous(range = c(1, 16)) +
  labs(title = "Industrias del Sector Intensivo (HS2)",
       subtitle = "Importaciones de China, Vietnam, Camboya, Malasia, Indonesia (2023)",
       x = "PCI",
       y = "Razón con respecto al Total de Importaciones [%]",
       size = "Total export value (Million USD)",
       fill = "Cluster") +
  theme_minimal()  +
  guides(fill=guide_legend(override.aes=list(size=6))) + 
  geom_hline(yintercept=50,  color = "red", size = 1.5) + 
  geom_vline(xintercept = ECI_pais, color = "black", size=1.5)

ggsave("../GTM/hs2_export_value_total_intensivo.png",dpi = 300,  width = 1167/90, height = 575/90,  bg = 'white')


### EXTENSIVO

### PCI VS SHARE ON TOTAL IMPORTS
## Export value to USA
ggplot(data_extensivo, 
       aes(x = pci, y = share_on_total_imports, size = export_value_to_usa, fill = product_hs92_name_1d, label = product_hs92_name_2d)) +
  geom_point(alpha = .5, 
             shape = 21) +
  geom_text(nudge_x = 0.1, nudge_y = 0.1, size = 4, color = "black")  +
  scale_size_continuous(range = c(1, 16)) +
  labs(title = "Industrias del Sector Extensivo (HS2)",
       subtitle = "Importaciones de China, Vietnam, Camboya, Malasia, Indonesia (2023)",
       x = "PCI",
       y = "Razón con respecto al Total de Importaciones [%]",
       size = "Export value to USA (Million USD)",
       fill = "Cluster") +
  theme_minimal()  +
  guides(fill=guide_legend(override.aes=list(size=6))) + 
  geom_hline(yintercept=50,  color = "red", size = 1.5) + 
  geom_vline(xintercept = -0.19, color = "black", size=1.5)

ggsave("../GTM/hs2_export_value_to_usa_extensivo.png",dpi = 300,  width = 1167/90, height = 575/90,  bg = 'white')

## Total export value
ggplot(data_extensivo, 
       aes(x = pci, y = share_on_total_imports, size = export_value, fill = product_hs92_name_1d, label = product_hs92_name_2d)) +
  geom_point(alpha = .5, 
             shape = 21) +
  geom_text(nudge_x = 0.1, nudge_y = 0.1, size = 4, color = "black")  +
  scale_size_continuous(range = c(1, 16)) +
  labs(title = "Industrias del Sector Extensivo (HS2)",
       subtitle = "Importaciones de China, Vietnam, Camboya, Malasia, Indonesia (2023)",
       x = "PCI",
       y = "Razón con respecto al Total de Importaciones [%]",
       size = "Total export value (Million USD)",
       fill = "Cluster") +
  theme_minimal()  +
  guides(fill=guide_legend(override.aes=list(size=6))) + 
  geom_hline(yintercept=50,  color = "red", size = 1.5) + 
  geom_vline(xintercept = ECI_pais, color = "black", size=1.5)

ggsave("../GTM/hs2_export_value_total_extensivo.png",dpi = 300,  width = 1167/90, height = 575/90,  bg = 'white')

### Distance VS SHARE ON TOTAL IMPORTS

## Export value to USA
ggplot(data_extensivo, 
       aes(x = distance, y = share_on_total_imports, size = export_value_to_usa, fill = product_hs92_name_1d, label = product_hs92_name_2d)) +
  geom_point(alpha = .5, 
             shape = 21) +
  geom_text(nudge_x = 0.001, nudge_y = 0.1, size = 3, color = "black")  +
  scale_size_continuous(range = c(1, 16)) +
  labs(title = "Industrias del Sector Extensivo (HS2)",
       subtitle = "Importaciones de China, Vietnam, Camboya, Malasia, Indonesia (2023)",
       x = "Distance",
       y = "Razón con respecto al Total de Importaciones [%]",
       size = "Export value to USA (Million USD)",
       fill = "Cluster") +
  theme_minimal()  +
  guides(fill=guide_legend(override.aes=list(size=6))) + 
  geom_hline(yintercept=30,  color = "red", size = 1.5, alpha = 0.3) + 
  geom_vline(xintercept = 0.8, color = "black", size=1.5, alpha = 0.3)

ggsave("../GTM/hs2_export_value_to_usa_extensivo_distance.png",dpi = 300,  width = 1167/90, height = 575/90,  bg = 'white')

## Total export value
ggplot(data_extensivo, 
       aes(x = distance, y = share_on_total_imports, size = export_value, fill = product_hs92_name_1d, label = product_hs92_name_2d)) +
  geom_point(alpha = .5, 
             shape = 21) +
  geom_text(nudge_x = 0.001, nudge_y = 0.1, size = 3, color = "black")  +
  scale_size_continuous(range = c(1, 16)) +
  labs(title = "Industrias del Sector Extensivo (HS2)",
       subtitle = "Importaciones de China, Vietnam, Camboya, Malasia, Indonesia (2023)",
       x = "Distance",
       y = "Razón con respecto al Total de Importaciones [%]",
       size = "Total export value (Million USD)",
       fill = "Cluster") +
  theme_minimal()  +
  guides(fill=guide_legend(override.aes=list(size=6))) + 
  geom_hline(yintercept=30,  color = "red", size = 1.5, alpha = 0.3) + 
  geom_vline(xintercept = 0.8, color = "black", size=1.5, alpha = 0.3)

ggsave("../GTM/hs2_export_value_total_extensivo_distance.png",dpi = 300,  width = 1167/90, height = 575/90,  bg = 'white')

### INTENSIVO HS4
## Export value to USA
ggplot(data_intensivo_4d, 
       aes(x = pci, y = share_on_total_imports, size = export_value_to_usa, fill = product_hs92_name_1d)) +
  geom_point(alpha = .5, 
             shape = 21) +
  scale_size_continuous(range = c(1, 16)) +
  labs(title = "Industrias del Sector Intensivo (HS4)",
       subtitle = "Importaciones de China, Vietnam, Camboya, Malasia, Indonesia (2023)",
       x = "PCI",
       y = "Razón con respecto al Total de Importaciones [%]",
       size = "Export value to USA (Million USD)",
       fill = "Cluster") +
  theme_minimal()  +
  guides(fill=guide_legend(override.aes=list(size=6))) + 
  geom_hline(yintercept=40,  color = "red", size = 1.5) + 
  geom_vline(xintercept = ECI_pais, color = "black", size=1.5)

## Total export value
ggplot(data_intensivo_4d, 
       aes(x = pci, y = share_on_total_imports, size = export_value, fill = product_hs92_name_1d)) +
  geom_point(alpha = .5, 
             shape = 21) +
  scale_size_continuous(range = c(1, 16)) +
  labs(title = "Industrias del Sector Intensivo (HS4)",
       subtitle = "Importaciones de China, Vietnam, Camboya, Malasia, Indonesia (2023)",
       x = "PCI",
       y = "Razón con respecto al Total de Importaciones [%]",
       size = "Total export value (Million USD)",
       fill = "Cluster") +
  theme_minimal()  +
  guides(fill=guide_legend(override.aes=list(size=6))) + 
  geom_hline(yintercept=40,  color = "red", size = 1.5) + 
  geom_vline(xintercept = ECI_pais, color = "black", size=1.5)

### Etiquetamos puntos en el cuadrante de interés
labeled_df = subset(data_intensivo_4d, share_on_total_imports>=30 & pci > -0.19)

## Export value to USA
ggplot(data_intensivo_4d, 
       aes(x = pci, y = share_on_total_imports, size = export_value_to_usa, fill = product_hs92_name_1d)) +
  geom_point(alpha = .5, 
             shape = 21) +
  scale_size_continuous(range = c(1, 16)) +
  labs(title = "Industrias del Sector Intensivo (HS4)",
       subtitle = "Importaciones de China, Vietnam, Camboya, Malasia, Indonesia (2023)",
       x = "PCI",
       y = "Razón con respecto al Total de Importaciones [%]",
       size = "Export value to USA (Million USD)",
       fill = "Cluster") +
  theme_minimal()  +
  guides(fill=guide_legend(override.aes=list(size=6))) + 
  geom_hline(yintercept=30,  color = "red", size = 1.5, alpha = 0.3) + 
  geom_vline(xintercept = ECI_pais, color = "black", size=1.5, alpha = 0.3)+ 
  geom_text_repel(data = labeled_df, aes(label = product_hs92_name_4d), size = 4) 

ggsave("../GTM/hs4_intensivo_export_value_to_usa.png",dpi = 300,  width = 1167/90, height = 575/90,  bg = 'white')


## Total export value
ggplot(data_intensivo_4d, 
       aes(x = pci, y = share_on_total_imports, size = export_value, fill = product_hs92_name_1d)) +
  geom_point(alpha = .5, 
             shape = 21) +
  scale_size_continuous(range = c(1, 16)) +
  labs(title = "Industrias del Sector Intensivo (HS4)",
       subtitle = "Importaciones de China, Vietnam, Camboya, Malasia, Indonesia (2023)",
       x = "PCI",
       y = "Razón con respecto al Total de Importaciones [%]",
       size = "Total export value (Million USD)",
       fill = "Cluster") +
  theme_minimal()  +
  guides(fill=guide_legend(override.aes=list(size=6))) + 
  geom_hline(yintercept=30,  color = "red", size = 1.5, alpha = 0.3) + 
  geom_vline(xintercept = ECI_pais, color = "black", size=1.5, alpha = 0.3)+ 
  geom_text_repel(data = labeled_df, aes(label = product_hs92_name_4d), size = 4) 

ggsave("../GTM/hs4_intensivo_export_value_total.png",dpi = 300,  width = 1167/90, height = 575/90,  bg = 'white')
